src/elm/Pages/Pipeline/View.elm

```diff
@@ -60,70 +67,71 @@ import Vela
 
 {-| viewPipeline : takes model and renders collapsible template previews and the pipeline configuration file for the desired ref.
 -}
-viewPipeline : PartialModel a -> Html Msg
-viewPipeline model =
+viewPipeline : PartialModel a -> Msgs msg -> Maybe Ref -> Html msg
+viewPipeline model msgs ref =
     div [ class "pipeline" ]
-        [ viewPipelineTemplates model.templates
-        , viewPipelineConfiguration model
+        [ viewPipelineTemplates model.templates msgs.showHideTemplates
+        , viewPipelineConfiguration model msgs ref
         ]
 
 
 {-| viewPipelineTemplates : takes templates and renders a list above the pipeline configuration.
 -}
-viewPipelineTemplates : ( WebData Templates, Error ) -> Html Msg
-viewPipelineTemplates templates =
-    case templates of
-        ( NotAsked, _ ) ->
+viewPipelineTemplates : PipelineTemplates -> msg -> Html msg
+viewPipelineTemplates { data, error, show } showHide =
+    case data of
+        NotAsked ->
             text ""
 
-        ( Loading, _ ) ->
+        Loading ->
             Util.smallLoaderWithText "loading pipeline templates"
 
-        ( Success t, _ ) ->
-            viewTemplates t
+        Success t ->
+            viewTemplates t show showHide
 
-        ( Failure _, err ) ->
-            viewTemplatesError err
+        Failure _ ->
+            viewTemplatesError error show showHide
 
 
 {-| viewTemplates : takes templates and renders a list of templates.
 
     Does not show if no templates are used in the pipeline.
 
 -}
-viewTemplates : Templates -> Html Msg
-viewTemplates templates =
+viewTemplates : Templates -> Bool -> msg -> Html msg
+viewTemplates templates open showHide =
     if not <| Dict.isEmpty templates then
         templates
             |> Dict.toList
             |> List.map viewTemplate
-            |> viewTemplatesDetails (class "-success")
+            |> viewTemplatesDetails (class "-success") open showHide
 
     else
         text ""
 
 
 {-| viewTemplates : renders an error from fetching templates.
 -}
-viewTemplatesError : Error -> Html Msg
-viewTemplatesError err =
+viewTemplatesError : Error -> Bool -> msg -> Html msg
+viewTemplatesError err open showHide =
     [ text <| "There was a problem fetching templates for this pipeline configuration"
     , div [ Util.testAttribute "pipeline-templates-error" ] [ text err ]
     ]
-        |> viewTemplatesDetails (class "-error")
+        |> viewTemplatesDetails (class "-error") open showHide
 
 
 {-| viewTemplatesDetails : takes templates content and wraps it in a details/summary.
 -}
-viewTemplatesDetails : Html.Attribute Msg -> List (Html Msg) -> Html Msg
-viewTemplatesDetails cls content =
+viewTemplatesDetails : Html.Attribute msg -> Bool -> msg -> List (Html msg) -> Html msg
+viewTemplatesDetails cls open showHide content =
```

Comment:
controlling the templates details `open` attribute using application state, so we may preserve this value when using the Build nav tabs.
Otherwise, the DOM opens it by default every time the page loads.